samples/algorithms/PhaseFlip.qs

```diff
@@ -0,0 +1,160 @@
+
+        // Measure and reset the physical qubit before releasing it.
+        let result = M(physicalQubit);
+        Reset(physicalQubit);
+        return result;
+    }
+
+    /// # Summary
+    /// This operation sets the state of the given qubit such that
+    /// it will have a 20% likelihood of resulting in a `Zero` and
+    /// 80% likelihood of resulting in a `One` when measured in the
+    /// computational basis. The input qubit is expected to be in
+    /// the |0〉 state.
+    ///
+    /// # Input
+    /// ## q
+    /// The given qubit to be put into superposition. It is assumed that this
+    /// qubit is in its default |0〉 state.
+    operation SetSampleState(q : Qubit) : Unit {
+        let alpha = 0.20;
+        Ry(2.0 * ArcCos(Sqrt(alpha)), q);
+    }
+
+    /// # Summary
+    /// This operation takes the given `physicalQubit` state,
+    /// (α|0〉 + β|1〉) / √2, and encodes it in the `aux` qubits. This
+    /// encodes all the qubits into a single logical qubit whose state reflects
+    /// the state of the given `physicalQubit`: (α|+++〉 + β|---〉) / √2. Note
+    /// that in this phase-flip example, the logical state |0〉 corresponds to
+    /// the physical state |+++〉, and the logical state |1〉 corresponds to the
+    /// physical state |---〉.
+    ///
+    /// # Input
+    /// ## physicalQubit
+    /// The qubit whose state, (α|0〉 + β|1〉) / √2, is to be encoded in the
+    /// logical qubit.
+    ///
+    /// ## aux
+    /// The auxiliary qubits that will be used as part of the encoding. These
+    /// should be grouped with the `physicalQubit` to form the logical qubit.
+    operation EncodeAsLogicalQubit(physicalQubit : Qubit, aux : Qubit[]) : Unit is Adj {
+        ApplyToEachA(CNOT(physicalQubit, _), aux);
+
+        // We change the basis of the physical qubits so that the
+        // logical state |0〉 corresponds to the physical state |+++〉,
+        // and the logical state |1〉 corresponds to the physical state |---〉.
+        ChangeBasis([physicalQubit] + aux);
+    }
+
+    /// # Summary
+    /// Changes the basis of the given qubits by applying a Hadamard operation
+    /// to each of them.
+    ///
+    /// # Input
+    /// ## qs
+    /// The given qubits to change the basis of.
+    operation ChangeBasis(qs : Qubit[]) : Unit is Adj {
+        ApplyToEachA(H, qs);
+    }
+
+    /// # Summary
+    /// This operation detects and corrects a single phase-flip error for a logical
+    /// qubit encoded as three physical qubits. When finished, the given register
+    /// of qubits will be in the state: (α|000〉 + β|111〉) / √2.
+    ///
+    /// # Input
+    /// ## physicalQubits
+    /// The given register of three physical qubits representing a single logical qubit
+    /// having superposition (α|+++〉 + β|---〉) / √2.
+    /// This logical qubit can have up to one phase-flip error that will be corrected.
+    operation CorrectError(physicalQubits : Qubit[]) : Unit {
```

Comment:
Similar to my comment on the bit-flip code sample, I would suggest changing the name of `physicalQubits` to `logicalQubits` and adding a `Fact` statement checking that the size of the register is 3.

Reply:
done